src/test/java/org/elasticsearch/script/mustache/MustacheScriptEngineTest.java

```diff
@@ -98,14 +103,60 @@ public void testEscapeJson() throws IOException {
                 escaped.append(c);
             }
             StringWriter target = new StringWriter();
-            assertThat(escaped.toString(), equalTo(JsonEscapingMustacheFactory.escape(writer.toString(), target).toString()));
+            assertThat(escaped.toString(), equalTo(InternalMustacheFactory.escape(writer.toString(), target).toString()));
         }
     }
 
+    @Test @Repeat(iterations = 5)
+    public void testArrayAccess() throws Exception {
+        Map<String, Object> vars = new HashMap<>();
+        Object data = randomFrom(
+                new String[] { "foo", "bar" },
+                ImmutableList.of("foo", "bar"),
+                ImmutableSet.of("foo", "bar"));
```

Comment:
I understand how retrieving the n-th object of an array or a List works - as Sets don't define an order referring to the n-th element makes no sense, does it?